src/OpenTelemetry/Metrics/HistogramMeasurements.cs

```diff
@@ -0,0 +1,42 @@
+// <copyright file="HistogramMeasurements.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Metrics
+{
+    internal class HistogramMeasurements
```

Comment:
Wouldn't this be a good candidate for a struct?

Reply:
Since we use an array of `MetricPoint`, we want to reduce the amount of memory we allocate on heap. Using a struct here, would mean that we allocate the `Histogram` related fields even for `Counter`, `Gauge` etc. If we use a class, then we allocate memory for these fields only if it's a Histogram metric.